src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/controller/MaintainSupplierPricingController.kt

```diff
@@ -74,8 +80,21 @@ class MaintainSupplierPricingController(@Autowired val supplierPricingService: S
         val ids = agencyIds(moveId)
 
         val sitesAndPrice = supplierPricingService.getExistingSiteNamesAndPrice(supplier, ids.first, ids.second)
+        val startOfMonth = model.getAttribute(HtmlController.DATE_ATTRIBUTE) as LocalDate
+        val effectiveYear = effectiveYearForDate(startOfMonth)
 
         model.addAttribute("form", PriceForm(moveId, sitesAndPrice.third.pounds().toString(), sitesAndPrice.first, sitesAndPrice.second))
+        model.addAttribute("contractualYearStart", "${effectiveYear}")
+        model.addAttribute("contractualYearEnd", "${effectiveYear + 1}")
+
+        val from = model.getAttribute(HtmlController.PICK_UP_ATTRIBUTE)
+        val to = model.getAttribute(HtmlController.DROP_OFF_ATTRIBUTE)
+        val url = UriComponentsBuilder.fromUriString(HtmlController.SEARCH_JOURNEYS_RESULTS_URL)
+
+        from.takeUnless { it == "" }.apply { url.queryParam(HtmlController.PICK_UP_ATTRIBUTE, from) }
+        to.takeUnless { it == "" }.apply { url.queryParam(HtmlController.DROP_OFF_ATTRIBUTE, to) }
+
+        model.addAttribute("cancelLink", url.build().toUriString())
```

Comment:
And here, would be good to add test coverage to the unit tests for these attributes...